ocs_ci/ocs/cluster.py

```diff
@@ -884,15 +884,33 @@ def validate_cluster_on_pvc():
         pvc_names,
         constants.CEPH_ROOK_IO_PVC_LABEL,
     )
-    osd_pods = get_pod_name_by_pattern("rook-ceph-osd", ns, filter="prepare")
-    for ceph_pod in mon_pods + osd_pods:
-        out = run_cmd(f"oc -n {ns} get pods {ceph_pod} -o yaml")
-        out_yaml = yaml.safe_load(out)
-        for vol in out_yaml["spec"]["volumes"]:
-            if vol.get("persistentVolumeClaim"):
-                claimName = vol.get("persistentVolumeClaim").get("claimName")
-                logger.info(f"{ceph_pod} backed by pvc {claimName}")
-                assert claimName in pvc_names, "Ceph Internal Volume not backed by PVC"
+    retry_validation = True
```

Comment:
Is the [retry decorator](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/utility/retry.py) not sufficient for this use case?

Reply:
I wanted to do like that but then I added the logic to speed up the process to skip already validated pods.  I was thinking about how to change it here and came up with the other solution where I will pass this validated_pods to the function from outside, so the next iteration of re-try should be quicker and skip those already validated pods.